test/it/ServicesISpec.scala

```diff
@@ -29,26 +31,54 @@ import scala.util.Try
 class ServicesISpec extends PlaySpec with GuiceOneAppPerSuite with TryValues {
   private val statementParser = new AccessibilityStatementParser
 
-  "the configuration files" should {
-    val sourceConfig   = app.injector.instanceOf[SourceConfig]
+  "parsing the configuration files" should {
+    val sourceConfig = app.injector.instanceOf[SourceConfig]
     val servicesFinder = app.injector.instanceOf[ServicesFinder]
 
     servicesFinder.findAll().foreach { (service: String) =>
       val source = sourceConfig.statementSource(service)
-
       val statementTry = Try(statementParser.parseFromSource(source).valueOr(throw _))
 
-      s"include a correctly formatted accessibility statement yaml file for $service" in {
+      s"enforce a correctly formatted accessibility statement yaml file for $service" in {
         statementTry must be a 'success
       }
 
-      s"not contain missing milestones for $service" in {
+      s"enforce statement not contain missing milestones for $service" in {
         val statement: AccessibilityStatement = statementTry.get
-
         val hasMilestones = statement.milestones.getOrElse(Seq.empty).nonEmpty
-
         hasMilestones || statement.isNonCompliant || statement.isFullyCompliant must be(true)
       }
+
+      s"enforce serviceDomain in the format of aaaa.bbbb.cccc for $service" in {
+        val domainRegex = "([a-z0-9-]*[\\.]*)*[a-z0-9]*"
+        val statement: AccessibilityStatement = statementTry.get
+        statement.serviceDomain.matches(domainRegex) must be(true)
+      }
+
+      s"enforce serviceUrl starting with / for $service" in {
+        val statement: AccessibilityStatement = statementTry.get
+        statement.serviceUrl.startsWith("/") must be(true)
+      }
+    }
+  }
+
+  "the file names in the directory" should {
+    val appConfig = app.injector.instanceOf[AppConfig]
+
+    val servicesDirectoryPath =
+      new File(getClass.getClassLoader.getResource(appConfig.servicesDirectory).getPath)
+    val fileNames =
+      servicesDirectoryPath.listFiles().toSeq.filter(_.isFile).map(_.getName).sorted
+
+    fileNames.foreach { fileName =>
+      s"end in extension .yml for $fileName" in {
+        fileName.endsWith(".yml") must be(true)
+      }
+
+      s"contain lowercase, letter or hyphen only for $fileName" in {
+        val serviceName = fileName.split("\\.").head
+        serviceName.matches("([a-z0-9-]*)*[a-z0-9]*") must be(true)
```

Comment:
I feel this regex could be more targeted - but I don't know how helpful that would be.  The rules I'd choose would be:

- Must start with a block of 1+ letters/numbers
- Can contain any number of single hyphens followed by more blocks of 1+ letters/numbers

I'd personally take this moment to refine the regex but I don't have a clear business need for it.

Reply:
I would just check that the filename is included in the services returned by ServicesFinder. We already have a regex in there which is `([0-9a-z\-]+)(\.cy)?\.yml`